### PR TITLE
MA-3547 Make Graal default JS engine

### PR DESCRIPTION
**Pull request by steviec**
Graal is now the default engine, but we allow rhino if specified via `jsEngine: rhino` in the test config.

There are subtle differences in the Graal implementation that _could_ affect users in edge case situations. Here are the ones I discovered while updating the tests:
- correctly resolves `${parseInt(3)}` to `3` (rhino would resolve to `3.0`)
- preserves newlines in ENV vars passed in via CLI or YAML (rhino would strip those out)
- correctly escapes double slashes (rhino would run two passes of escaping, reducing `////` to `/`)

## Testing

- ran all tests including with `./gradlew build`
- did not run e2e tests (assuming CI will do that)

## Issues fixed
MA-3547
https://github.com/mobile-dev-inc/maestro/issues/2049

### CODE REVIEW

File: maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt [hunk 1]
```diff
@@ -290,13 +290,19 @@ class Orchestra(
...
             jsEngine.close()
         }
-        val shouldUseGraalJs =
-            config?.ext?.get("jsEngine") == "graaljs" || System.getenv("MAESTRO_USE_GRAALJS") == "true"
+        val isRhinoExplicitlyRequested = config?.ext?.get("jsEngine") == "rhino"
+        
+        // Warn users about deprecated Rhino JS engine
+        if (isRhinoExplicitlyRequested) {
+            logger.warn("⚠️  The Rhino JS engine (jsEngine: rhino) is deprecated and will be removed in a future version. Please migrate to GraalJS (the default) for better performance and compatibility.")
+        }
```
Comment (amanjeetsingh150) [inline maestro-orchestra/src/main/java/maestro/orchestra/Orchestra.kt:298]: I don't think this will be shown to users this just redirects to the maestro.log file on debug artifacts.

For showing to users we have Insights abstraction. You can use that to show this warning.
